tests/Optimizer/Transformer/MinifyHtmlTest.php

```diff
@@ -145,4 +147,46 @@ public function testTransform($source, $expectedHtml, $expectedErrors = [])
         $this->assertSimilarMarkup($expectedHtml, $document->saveHTML());
         $this->assertSameErrors($expectedErrors, $errors);
     }
+
+    /**
+     * Test optional amp-script minification.
+     *
+     * @covers \AmpProject\Optimizer\Transformer\MinifyHtml::minifyAmpScript()
+     */
+    public function testAmpScriptMinification()
+    {
+        $source = TestMarkup::DOCTYPE . '<html ⚡> <head> ' .
+            TestMarkup::META_CHARSET .
+            ' </head> <body>' .
+                "\n<script id=\"hello-world\" type=\"text/plain\" target=\"amp-script\">\n" .
+                "  const btn = document.querySelector('button');\n" .
+                "  btn.addEventListener('click', () => {\n" .
+                "    document.body.textContent = 'Hello World!';\n" .
+                "  });\n" .
+                "</script>\n" .
+            '</body> </html>';
+
+        $expectedHtml = TestMarkup::DOCTYPE . '<html ⚡><head>' .
+            TestMarkup::META_CHARSET .
+            '</head><body><script id="hello-world" type="text/plain" target="amp-script">' .
+            'const btn=document.querySelector(\'button\');btn.addEventListener' .
+            '(\'click\',()=>{document.body.textContent=\'Hello World!\';});' .
+            '</script></body></html>';
+
+        $document      = Document::fromHtml($source);
+        $configuration = new MinifyHtmlConfiguration([MinifyHtmlConfiguration::MINIFY_AMP_SCRIPT => true]);
+        $transformer   = new MinifyHtml($configuration);
+        $errors        = new ErrorCollection();
+
+        $transformer->transform($document, $errors);
+
+        if (! class_exists(Peast::class)) {
+            $expectedErrors = [
+                new MissingPackage('The optional package mck89/peast package is required to minify inline amp-script.'),
+            ];
+            $this->assertSameErrors($expectedErrors, $errors);
```

Comment:
I think we should rather skip the test here if the dependency is missing. Otherwise, we can never know whether the amp-script minification has been tested at all, as the test will always succeed either way.

In general, a mechanism like this (adapt the test based on conditions) should never be used, it is preferable to skip the test or have two separate tests instead.

```suggestion
            $this->markTestSkipped('The optional package mck89/peast is needed to test inline amp-script minification.');
```

Reply:
Got it. Thanks.

Reply:
@schlessera I've updated the test using separate tests - one for with Peast and another for without Peast. Without the test for the `MissingPackage` error, it is hard to achieve the perfect coverage score. Please review and let me know your feedback on this.